### DISCUSSION TITLE
Access to Adroid storage ,

### DISCUSSION THREAD

**Discussion by jk-2022** [flet-dev/flet#2341] [General]
How can i access to android external storage?

**Comment by ndonkoHenri** [comment]
https://github.com/flet-dev/flet/issues/2357

**Comment by Vallekralle** [comment]
**Hi jk-2022**,

I think you can solve your problem with the **FilePicker** [docs](https://flet.dev/docs/guides/python/file-picker-and-uploads/). It is a class from **flet**, so you can import it just like you would do it e.g. for a Textfield or ElevatedButton etc., you don't have to install anything. From there you can choose between _3 different methods_: 

**pick_files()
save_file()
get_directory_path()**

You can call one of these methods with an on_click-event (e.g. from an ElevatedButton, like mentioned before...) or how ever you want.
After that you have to implement a callback-function, it gets called when the user has selected his file or folder etc. The method will need one parameter: "e: ft.FilePickerResultEvent". With this you can implement your logic in the method like, e.path (to get the path) or e.files (to get the files).

I hope I could help you a little, **don't forget to read the docs they are very helpful and maybe more understandable**,
Stay safe and have fun coding!

**_Example:_**
```
import flet as ft
from flet import ElevatedButton, ControlEvent, FilePicker, FilePickerResultEvent, Row


def main(page: ft.Page):
    page.title = "Flet FilePicker"
    page.vertical_alignment = ft.MainAxisAlignment.CENTER

    def open_folder(e: ControlEvent):
        file_picker.get_directory_path()  # If you want to select multiple files change to "file_picker.pick_files(allow_multiple=True)"

    def on_dialog_result(e: FilePickerResultEvent):
        print(e.path)

    file_picker = FilePicker(on_result=on_dialog_result)
    page.overlay.append(file_picker)
    page.update()

    file_picker_btn: ElevatedButton = ElevatedButton(
        text="Folder", icon=ft.icons.FOLDER, on_click=open_folder
    )

    page.add(
        Row(
            [
                file_picker_btn
            ], alignment=ft.MainAxisAlignment.CENTER
        )
    )


if __name__ == "__main__":
    ft.app(target=main)

```
